libs/groovylib/src/mill/groovylib/TestGroovyMavenModule.scala

```diff
@@ -0,0 +1,25 @@
+package mill.groovylib
+
+import mill.*
+
+/**
+ * A [[GroovyModule]] for tests with a Maven compatible directory layout:
+ * `src/test/groovy`, `src/test/resources`, etc.
+ * This is useful when only the tests are written in Groovy while the production code remains in Java or other.
+ *
+ * Requires an outer module (like JavaModule) otherwise [[moduleDir]] must be overridden manually.
+ */
+trait TestGroovyMavenModule extends GroovyMavenModule {
+
+  /**
+   * The name of this module's folder within `src/`: e.g. `src/test/`, `src/integration/`,
+   * etc. Defaults to the name of the module object, but can be overridden by users
+   */
+  def testModuleName: String = moduleCtx.segments.last.value
+
+  private def testSources = Task.Sources(moduleDir / os.up / "src" / testModuleName / "groovy")
+  override def sources: T[Seq[PathRef]] = testSources()
+
+  private def testResources = Task.Sources(moduleDir / os.up / "src" / testModuleName / "resources")
+  override def resources: T[Seq[PathRef]] = testResources()
```

Comment:
TBH, I don't like this. This _assumes_, the outer module is located at one level above, but effectively, Mill does not enforce it, so this is potentially using the wrong sources or it blows up.

To properly set it up, you might need to define a `JavaWithGroovyTestModule` with an inner test trait.

```scala
trait JavaMavenModuleWithGroovyTests extends JavaMavenModule { outer =>
  trait GroovyMavenTests extends JavaMavenTests {
    ...
  }
}
```

Reply:
Yeah, I didn't like this either but the only reason where you need this is when you already define an outer module which has another language-module and only your test sources differ. In case you need a pure Groovy test-module you can just use GroovyMavenModule.

I will try your suggestion to get rid of the assumption going up a level.